src/librbd/cache/rwl/Types.h

```diff
@@ -184,26 +190,33 @@ class DeferredContexts {
 };
 
 /* Pmem structures */
+#ifdef WITH_RBD_RWL
 POBJ_LAYOUT_BEGIN(rbd_rwl);
 POBJ_LAYOUT_ROOT(rbd_rwl, struct WriteLogPoolRoot);
 POBJ_LAYOUT_TOID(rbd_rwl, uint8_t);
 POBJ_LAYOUT_TOID(rbd_rwl, struct WriteLogPmemEntry);
 POBJ_LAYOUT_END(rbd_rwl);
+#endif
 
 struct WriteLogPmemEntry {
```

Comment:
Nit: perhaps you should have a generic base class `WriteLogEntry` and then you have a derived `WriteLogPmemEntry` that adds the pmdk-specific things. Same comment for `WriteLogPoolRoot`.

Reply:
Do you mean have a generic base `WriteLogEntry` class and have `PmemEntry` and `SSDEntry` derive from it? The reason they haven't been split up is because in LogEntry class [0], it's easier to access them with a single type entry, as it's interchangeably used in a lot of places in SSD and RWL implementation. If we derive from a base class, derived class variables won't be accessible in base class type (aka can't do Base a = new Derived() since there are no virtual variables in c++)

[0] https://github.com/ceph/ceph/blob/master/src/librbd/cache/rwl/LogEntry.h#L26-L27